syntax = "proto3";

// Market service definition
service MarketSimulation {
    // Sends generated market data back, whether historical or randomly generated data.
    rpc CreateMarketStream (MarketDataRequest) returns (stream MarketDataResponse) {}
}

// The response message containing a ticker and the security's floating point value.
message MarketDataResponse {
    string ticker = 1;
    float value = 2;
}

// The request message containing a ticker and the security's starting floating point value.
message MarketDataRequest {
    string ticker = 1;
    float starting_value = 2;
    int32 num_tests = 3;
}